src/main/java/com/rabbitmq/client/impl/MetricsStatistics.java

```diff
@@ -21,128 +53,306 @@
     private final Meter rejectedMessages;
 
 
-    public MetricsStatistics(MetricRegistry registry) {
+    public MetricsStatistics(MetricRegistry registry, String metricsPrefix) {
         this.registry = registry;
-        this.connections = registry.counter("connections");
-        this.channels = registry.counter("channels");
-        this.publishedMessages = registry.meter("published");
-        this.consumedMessages = registry.meter("consumed");
-        this.acknowledgedMessages = registry.meter("acknowledged");
-        this.rejectedMessages = registry.meter("rejected");
+        this.connections = registry.counter(metricsPrefix+".connections");
+        this.channels = registry.counter(metricsPrefix+".channels");
+        this.publishedMessages = registry.meter(metricsPrefix+".published");
+        this.consumedMessages = registry.meter(metricsPrefix+".consumed");
+        this.acknowledgedMessages = registry.meter(metricsPrefix+".acknowledged");
+        this.rejectedMessages = registry.meter(metricsPrefix+".rejected");
     }
 
     public MetricsStatistics() {
         this(new MetricRegistry());
     }
+    
+    public MetricsStatistics(MetricRegistry metricRegistry) {
+        this(metricRegistry, "rabbitmq");
+    }
 
     @Override
-    public long getConnectionCount() {
-        return connections.getCount();
+    public void newConnection(final Connection connection) {
+        try {
+            if(connection.getId() == null) {
+                connection.setId(UUID.randomUUID().toString());
+            }
+            connections.inc();
+            connectionState.put(connection.getId(), new ConnectionState(connection));
+            connection.addShutdownListener(new ShutdownListener() {
```

Comment:
We need to make sure that we don't accumulate more than one listener between connection recoveries.